src/MOI_wrapper.jl

```diff
@@ -0,0 +1,2708 @@
+]
+
+function MOI.get(model::Optimizer, ::MOI.RawStatusString)
+    status_code = get_status_code(model.inner)
+    if 1 <= status_code <= length(RAW_STATUS_STRINGS)
+        return RAW_STATUS_STRINGS[status_code][2]
+    end
+    return MOI.OTHER_ERROR
+end
+
+function MOI.get(model::Optimizer, ::MOI.TerminationStatus)
+    status_code = get_status_code(model.inner)
+    if 1 <= status_code <= length(RAW_STATUS_STRINGS)
+        return RAW_STATUS_STRINGS[status_code][1]
+    end
+    return MOI.OTHER_ERROR
+end
+
+function MOI.get(model::Optimizer, ::MOI.PrimalStatus)
+    stat = get_status(model.inner)
+    if stat == :optimal
+        return MOI.FEASIBLE_POINT
+    elseif stat == :solution_limit
+        return MOI.FEASIBLE_POINT
+    elseif (stat == :inf_or_unbd || stat == :unbounded) && _has_primal_ray(model)
+        return MOI.INFEASIBILITY_CERTIFICATE
+    elseif stat == :suboptimal
+        return MOI.FEASIBLE_POINT
+    elseif is_mip(model.inner) && get_sol_count(model.inner) > 0
+        return MOI.FEASIBLE_POINT
+    end
+    return MOI.NO_SOLUTION
+end
+
+function _has_dual_ray(model::Optimizer)
+    try
+        # Note: for performance reasons, we try to get 1 element because for
+        # some versions of Gurobi, we cannot query 0 elements without error.
+        get_dblattrarray(model.inner, "FarkasDual", 1, 1)
+        return true
+    catch ex
+        if isa(ex, GurobiError)
+            return false
+        else
+            rethrow(ex)
+        end
+    end
+end
+
+function MOI.get(model::Optimizer, ::MOI.DualStatus)
+    stat = get_status(model.inner)
+    if is_mip(model.inner)
+        return MOI.NO_SOLUTION
+    elseif is_qcp(model.inner) && MOI.get(model, MOI.RawParameter("QCPDual")) != 1
+        return MOI.NO_SOLUTION
+    elseif stat == :optimal
+        return MOI.FEASIBLE_POINT
+    elseif stat == :solution_limit
+        return MOI.FEASIBLE_POINT
+    elseif (stat == :inf_or_unbd || stat == :infeasible) && _has_dual_ray(model)
+        return MOI.INFEASIBILITY_CERTIFICATE
+    elseif stat == :suboptimal
+        return MOI.FEASIBLE_POINT
+    end
+    return MOI.NO_SOLUTION
+end
+
+function _has_primal_ray(model::Optimizer)
+    try
+        # Note: for performance reasons, we try to get 1 element because for
+        # some versions of Gurobi, we cannot query 0 elements without error.
+        get_dblattrarray(model.inner, "UnbdRay", 1, 1)
+        return true
+    catch ex
+        if isa(ex, GurobiError)
+            return false
+        else
+            rethrow(ex)
+        end
+    end
+end
+
+function MOI.get(model::Optimizer, ::MOI.VariablePrimal, x::MOI.VariableIndex)
+    if model.has_unbounded_ray
+        return get_dblattrelement(model.inner, "UnbdRay", _info(model, x).column)
+    else
+        return get_dblattrelement(model.inner, "X", _info(model, x).col
```

Comment:
Do you think it could be worthwile to add a bridge `SOCToQuadBridge` in MOI?

Reply:
Not sure, Xpress supports both SOC and Quad and we could get to a point of supporting both